libs/filabridge/include/private/filament/Variant.h

```diff
@@ -53,144 +53,133 @@ namespace filament {
         //
         // Standard variants:
         //                      +-----+-----+-----+-----+-----+-----+-----+-----+
-        //                      |  0  | VSM | FOG |  0  | SKN | SRE | DYN | DIR |    40 (-24)
+        //                      |  0  | VSM | FOG |  0  | SKN | SRE | DYN | DIR |    64 - 24 = 40
         //                      +-----+-----+-----+-----+-----+-----+-----+-----+
         //      Vertex shader            0     0     0     X     X     X     X
         //    Fragment shader            X     X     0     0     X     X     X
-        //           Reserved            X     X     0     X     1     0     0
-        //           Reserved            1     X     0     X     0     X     X
+        //           Reserved            X     X     0     X     1     0     0      [ -8]
+        //           Reserved            1     X     0     X     0     X     X      [-16]
         //
         // Depth variants:
         //                      +-----+-----+-----+-----+-----+-----+-----+-----+
-        //                      |  0  | VSM | PCK |  1  | SKN |  0  |  0  |  0  |     6 (-58)
+        //                      |  0  | VSM | PCK |  1  | SKN |  0  |  0  |  0  |   8 - 2 = 6
         //                      +-----+-----+-----+-----+-----+-----+-----+-----+
         //       Vertex depth            X     0     1     X     0     0     0
         //     Fragment depth            X     X     1     0     0     0     0
-        //           Reserved            1     1     1     X     0     0     0
-        //           Reserved            X     X     1     X     0     0     1
-        //           Reserved            X     X     1     X     0     1     0
-        //           Reserved            X     X     1     X     0     1     1
-        //           Reserved            X     X     1     X     1     0     0
-        //           Reserved            X     X     1     X     1     0     1
-        //           Reserved            X     X     1     X     1     1     0
-        //           Reserved            X     X     1     X     1     1     1
+        //           Reserved            1     1     1     X     0     0     0     [ -2]
         //
-        // 46 variants used, 82 reserved
+        // 46 variants used, 82 reserved (128 - 46)
         //
+        // note: a valid variant can be neither a valid vertex nor a valid fragment variant
```

Comment:
This note confuses me. Do you mean a "_reserved_ variant can be neither..."

Reply:
No, I meant that a valid variant (i.e. `!isReserved()`) can be invalid as both a fragment and vertex variant. This is counter intuitive and that's why I noted it here.

For e.g. a variant that as FOG|SKN is absolutely valid. But, it's not a valid fragment variant nor a valid vertex variant. This is because fragment variants don't allow SKN and vertex variants don't allow FOG.

Reply:
I see. I would replace "can be" with "could potentially be".

I think the usage of "valid" to mean "not reserved" is confusing, since some "valid" variants will still get filtered out. Maybe we should add a note about that.